application-ckeditor-plugins/src/main/resources/xwiki-image/imageSelector.js

```diff
@@ -167,4 +122,10 @@ define('imageSelector', ['jquery', 'modal', 'resource', 'l10n!imageSelector'],
         });
       }
     });
+    return {
+      createModal: createModal,
+      saveSelectedImageReference: saveSelectedImageReference,
+      removeSelectedImageReference: removeSelectedImageReference,
```

Comment:
I think it makes more sense to have a single method that receives a **list** of image references. Of course, we would take into account only the first item for now, but this allows us to add support for multiple selection later without modifying the API. Also, passing an empty list would mean clearing the selection.

Reply:
I propose `updateSelectedImageReference`